### PR TITLE
wget implemented

### PR DESCRIPTION
**Pull request by huseyingokay**

### CODE REVIEW

File: run_idempotent_azure.sh [hunk 1]
```diff
@@ -36,16 +36,25 @@ modified_slug_module="${modifiedslug_with_sha}=${modified_module}"
 MVNOPTIONS="-Ddependency-check.skip=true -Dmaven.repo.local=$AZ_BATCH_TASK_WORKING_DIR/dependencies_${modifiedslug_with_sha}=${modified_module} -Dgpg.skip=true -DfailIfNoTests=false -Dskip.installnodenpm -Dskip.npm -Dskip.yarn -Dlicense.skip -Dcheckstyle.skip -Drat.skip -Denforcer.skip -Danimal.sniffer.skip -Dmaven.javadoc.skip -Dfindbugs.skip -Dwarbucks.skip -Dmodernizer.skip -Dimpsort.skip -Dmdep.analyze.skip -Dpgpverify.skip -Dxml.skip -Dcobertura.skip=true -Dfindbugs.skip=true"
 
 # echo "================Cloning the project"
-bash $dir/clone-project.sh "$slug" "$modified_slug_module" "$input_container"
+bash $dir/clone-project.sh "$slug" "${modifiedslug_with_sha}=${modified_module}" "$input_container"
```
Comment (winglam) [inline run_idempotent_azure.sh]: Why is this line changed? Should be fine to just use $modified_slug_module

File: run_idempotent_azure.sh [hunk 2]
```diff
@@ -36,16 +36,25 @@ modified_slug_module="${modifiedslug_with_sha}=${modified_module}"
...
-    exit 1
+        cd ~/
+        wget "https://github.com/$slug/archive/$sha".zip
+        ret=${PIPESTATUS[0]}
+        if [[ $ret != 0 ]]; then
+            echo "$line,${modifiedslug_with_sha}=${modified_module},failed_wget" >> $AZ_BATCH_TASK_WORKING_DIR/$input_container/results/"${modifiedslug_with_sha}=${modified_module}-results".csv
+            echo "Compilation failed. Actual: $ret"
+            exit 1
+        else
+            echo "$line,${modifiedslug_with_sha}=${modified_module},passed_wget" >> $AZ_BATCH_TASK_WORKING_DIR/$input_container/results/"${modifiedslug_with_sha}=${modified_module}-results".csv
```
Comment (winglam) [inline run_idempotent_azure.sh]: This is not enough for passed_wget. We did not install the project yet! You can set a variable if wget succeeded. If wget failed then you should be using ```cannot_checkout_or_wget```. AFTER install-project.sh is run you can then check whether wget succeeded or not. If wget succeeded then if install project succeeded the result is passed_wget, otherwise install project failed and the result is failed_wget.

File: run_idempotent_azure.sh [hunk 3]
```diff
@@ -36,16 +36,25 @@ modified_slug_module="${modifiedslug_with_sha}=${modified_module}"
 MVNOPTIONS="-Ddependency-check.skip=true -Dmaven.repo.local=$AZ_BATCH_TASK_WORKING_DIR/dependencies_${modifiedslug_with_sha}=${modified_module} -Dgpg.skip=true -DfailIfNoTests=false -Dskip.installnodenpm -Dskip.npm -Dskip.yarn -Dlicense.skip -Dcheckstyle.skip -Drat.skip -Denforcer.skip -Danimal.sniffer.skip -Dmaven.javadoc.skip -Dfindbugs.skip -Dwarbucks.skip -Dmodernizer.skip -Dimpsort.skip -Dmdep.analyze.skip -Dpgpverify.skip -Dxml.skip -Dcobertura.skip=true -Dfindbugs.skip=true"
 
 # echo "================Cloning the project"
-bash $dir/clone-project.sh "$slug" "$modified_slug_module" "$input_container"
+bash $dir/clone-project.sh "$slug" "${modifiedslug_with_sha}=${modified_module}" "$input_container"
 ret=${PIPESTATUS[0]}
 if [[ $ret != 0 ]]; then
     if [[ $ret == 2 ]]; then
         echo "$line,${modifiedslug_with_sha}=${modified_module},cannot_clone" >> $AZ_BATCH_TASK_WORKING_DIR/$input_container/results/"${modifiedslug_with_sha}=${modified_module}-results".csv
+        echo "Couldn't download the project. Actual: $ret"
```
Comment (winglam) [inline run_idempotent_azure.sh:44]: You have also changed this now so that if the $ret is not 1 or 2 then it would not exit the script, which is not desireable. Please add an else clause for
```
echo "Compilation failed. Actual: $ret"
exit 1
```

File: run_idempotent_azure.sh (hunk 2 above)
Comment (winglam) [reply run_idempotent_azure.sh]: For this else clause specifically, you can just echo a message such as ```echo "git checkout failed but wget successfully downloaded the project and sha, proceeding to the rest of this script"```

**Review by winglam — COMMENTED** [review]
Make sure you change all scripts that have the wget commands according to the comments I provided pelase

File: run_idempotent_azure.sh (hunk 1 above)
Comment (huseyingokay) [reply run_idempotent_azure.sh]: Actually I should have added module='.' and modified-module at the top of MVNOPTIONS and I realized that I added the default module='.' to run_cloc_azure.sh. That means this modified line was missing before and I corrected right now :(

Comment (winglam) [reply run_idempotent_azure.sh]: I still don't understand why you had to change this line in this file and in many other files.
$modified_slug_module is the same as ${modifiedslug_with_sha}=${modified_module} so we should be able to use $modified_slug_module

In fact, even in MVNOPTIONS we should be able to directly use $modified_slug_module instead of ${modifiedslug_with_sha}=${modified_module}

Comment (huseyingokay) [reply run_idempotent_azure.sh]: I have mixed the words, fixed right now

**Review by winglam — COMMENTED** [review]
LGTM!
